### PR TITLE
CI: update to newer GA Fedora and RHEL runners

### PR DESCRIPTION
**Pull request by jrusz**

### CODE REVIEW

**Comment by jrusz** [comment]
I wonder, does it make sense to add RHEL-9 GA here? @gicmo or @ondrejbudai would you happen to know?

**Comment by ondrejbudai** [comment]
I don't think so, `koji-osbuild` is not yet built for RHEL 9 anywhere I think.

**Comment by jrusz** [comment]
Podman v4 brought some breaking changes in the `network` options so this will need some updates https://github.com/containers/podman/releases/tag/v4.0.0 

Also should I just try to fix all of those pylint issues @ondrejbudai ?

**Comment by ondrejbudai** [comment]
> Podman v4 brought some breaking changes in the `network` options so this will need some updates https://github.com/containers/podman/releases/tag/v4.0.0

Were you able to fix them?

> Also should I just try to fix all of those pylint issues @ondrejbudai ?

Yeah, makes sense, otherwise we would need to disable them.

**Comment by jrusz** [comment]
Had to tweak it a little bit, should be fine now

File: test/run-builder.sh [hunk 1]
```diff
@@ -22,8 +22,22 @@ else
 fi
 
 builder_start() {
-  GATEWAY_IP=$(${CONTAINER_RUNTIME} network inspect org.osbuild.koji | jq -r ".[0].plugins[0].ipam.ranges[0][0].gateway")
+  set +e
+  # Podman v4 introduced breaking changes to network option
```
Comment (ondrejbudai) [inline test/run-builder.sh:26]: If we switched the CI to fedora 36 (which is what Fedora rel-eng is switching to slowly, we wouldn't probably need all of this, would that be possible?

Comment (jrusz) [reply test/run-builder.sh:26]: Yeah I thought about this option as well, but we don't have F-36 runners yet in `gitlab-ci-terraform` I'll add them then and ask for review there :)

Comment (jrusz) [reply test/run-builder.sh:26]: Adding runners in https://github.com/osbuild/gitlab-ci-terraform/pull/66

**Comment by jrusz** [comment]
I don't understand how this could have worked on fedora-35 ... If I take the osbuild commit SHA from `deploy.sh` and take a look at osbuild repo there was no rpmbuild ran for fedora-35...

**Comment by jrusz** [comment]
Hm, I guess it only requires osbuild-composer and that had rpmbuild ran. I'll go enable rpmbuild for both just to be sure and come back with new SHA here.

**Comment by jrusz** [comment]
Well, we got past the podman update with very little change needed, but it seems we found/introduced a new problem. Tests on both distros are failing....

**Comment by jrusz** [comment]
I guess it's caused by the updated podman version :/ ... It fails with some connection errors and happend with older osbuild and osbuild-compsoser packages too. I'll see if I can figure out what's wrong.

**Comment by jrusz** [comment]
@ondrejbudai so from my testing it seems it's just failing because of `osbuild-compsoer` version... It still works with v45 but starts failing with v46 + and I have no idea what could be wrong, could you help please? This is an example of the failure https://gitlab.com/redhat/services/products/image-builder/ci/koji-osbuild/-/jobs/2636441587

**Comment by ondrejbudai** [comment]
pushed a commit that might fix it, the field name was imho incorrect.

**Comment by jrusz** [comment]
I tried this one before and it failed.

**Comment by Gundersanne** [comment]
Using `typ` was a hack I suggested, the real solution is to update composer's mock-openid provider to also return a consistent token in case of a client-credentials type grant, which koji-osbuild is using.

File: test/data/osbuild-composer.toml [hunk 2]
```diff
@@ -14,3 +15,4 @@ keytab = "/etc/osbuild-composer/client.keytab"
 [worker]
 allowed_domains = ["localhost", "client.osbuild.local"]
 ca = "/etc/osbuild-composer/ca-crt.pem"
+jwt_tenant_provider_fields = ["rh-org-id"]
```
Comment (Gundersanne) [inline test/data/osbuild-composer.toml:18]: I don't think local worker + jwt works. You need a remote rework + all the jwt related config similar to the koji config section.

**Comment by ondrejbudai** [comment]
Superseded by https://github.com/osbuild/koji-osbuild/pull/104

**Pull request closed without merging**